watertap/unit_models/nanofiltration_0D.py

```diff
@@ -0,0 +1,675 @@
+    **PropertyParameterObject** - a PropertyParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_args",
+        ConfigDict(
+            implicit=True,
+            description="Arguments to use for constructing property packages",
+            doc="""A ConfigDict with arguments to be passed to a property
+    block(s) and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}""",
+        ),
+    )
+    CONFIG.declare(
+        "include_pressure_balance",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            description="Whether to include pressure balance for retentate side.",
+        ),
+    )
+    CONFIG.declare(
+        "has_retentate_pressure_drop",
+        ConfigValue(
+            default=False,
+            domain=Bool,
+            description="Whether to include pressure drop in the retentate side.",
+        ),
+    )
+    CONFIG.declare(
+        "include_temperature_equality",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            description="Whether to include temperature equality constraints",
+            doc="""Argument indicating whether temperature equality should be
+            inlcluded. If this is False, no energy balance constraint will be
+            written.""",
+        ),
+    )
+    CONFIG.declare(
+        "electroneutrality_ion",
+        ConfigValue(
+            default="Cl_-",
+            domain=str,
+            description="Balancing ion to use to maintain electroneutrality",
+            doc="""Name of ion to use to maintain electroneutrality in
+            permeate stream. If an ion is provided, the split fraction
+            for this ion will be solved implicitly to ensure electroneutrality
+            is maintained in the permeate stream. If None, user must provide
+            spl;it fractions for all monovalent ions and electroneutrality is
+            not guaranteed..""",
+        ),
+    )
+
+    def build(self):
+        super().build()
+
+        prop_params = self.config.property_package
+
+        # Check that property package only supports single phase
+        # TODO: Extend the model to handle phase equilibrium
+        if len(prop_params.phase_list) > 1:
+            raise ConfigurationError(
+                "Nanofiltration0D model only supports single phase "
+                "property packages."
+            )
+
+        # Check that the electroneutrality ion is a valid monovalent ion
+        if self.config.electroneutrality_ion is not None:
+            try:
+                bal_ion = prop_params.get_component(self.config.electroneutrality_ion)
+                # Check that balancing ion is monovalent
+                if (
+                    not hasattr(bal_ion.config, "charge")
+                    or abs(bal_ion.config.charge) > 1
```

Comment:
Mainly because this model was written with the assumption that higher valance species would be rejected (and thus are not suitable for charge balancing).